src/main/java/org/dswarm/graph/json/stream/ModelParser.java

```diff
@@ -0,0 +1,542 @@
+					"cannot parse Resource JSON, couldn't find statement array beginning; expected '[' but found '%s'", jp.getCurrentToken()),
+					jp.getCurrentLocation());
+		}
+
+		final Resource resource = new Resource(resourceURI);
+
+		while (jp.nextToken() != JsonToken.END_ARRAY) {
+
+			final Statement statement = parseStatement(jp);
+
+			resource.addStatement(statement);
+		}
+
+		if (jp.nextToken() != JsonToken.END_OBJECT) {
+
+			jp.close();
+
+			throw new JsonParseException(String.format("cannot parse Resource JSON, couldn't find ending; expected '}' but found '%s'",
+					jp.getCurrentToken()), jp.getCurrentLocation());
+		}
+
+		return resource;
+	}
+
+	private Statement parseStatement(final JsonParser jp) throws IOException {
+
+		if (!jp.hasCurrentToken()) {
+
+			jp.close();
+
+			throw new JsonParseException("cannot parse Statement JSON, stream is empty", jp.getCurrentLocation());
+		}
+
+		if (jp.getCurrentToken() != JsonToken.START_OBJECT) {
+
+			jp.close();
+
+			throw new JsonParseException(String.format("cannot parse Statement JSON, couldn't find beginning; expected '{' but found '%s'",
+					jp.getCurrentToken()), jp.getCurrentLocation());
+		}
+
+		Long id = null;
+		String uuid = null;
+		Node subject = null;
+		Predicate predicate = null;
+		Node object = null;
+		Long order = null;
+		String evidence = null;
+		String confidence = null;
+
+		while (jp.nextToken() != JsonToken.END_OBJECT) {
+
+			if (jp.getCurrentToken() == JsonToken.FIELD_NAME) {
+
+				final String nextField = jp.getCurrentName();
+
+				switch (nextField) {
+
+					case ModelStatics.ID_IDENTIFIER:
+
+						id = getLongValue(jp);
+
+						break;
+					case ModelStatics.UUID_IDENTIFIER:
+
+						uuid = getStringValue(jp);
+
+						break;
+					case ModelStatics.SUBJECT_IDENTIFIER:
+
+						subject = getSubjectNode(jp, nextField);
+
+						break;
+					case ModelStatics.PREDICATE_IDENTIFIER:
+
+						predicate = getPredicate(jp, nextField);
+
+						break;
+					case ModelStatics.OBJECT_IDENTIFIER:
+
+						object = getObjectNode(jp, nextField);
+
+						break;
+					case ModelStatics.ORDER_IDENTIFIER:
+
+						order = getLongValue(jp);
+
+						break;
+					case ModelStatics.EVIDENCE_IDENTIFIER:
+
+						evidence = getStringValue(jp);
+
+						break;
+					case ModelStatics.CONFIDENCE_IDENTIFIER:
+
+						confidence = getStringValue(jp);
+
+						break;
+					default:
+
+						jp.close();
+
+						throw new JsonParseException(String.format(
+								"unexpected JSON token; expected one of '%s','%s','%s','%s','%s','%s','%s','%s', for this statement, but found '%s'",
+								ModelStatics.ID_IDENTIFIER, ModelStatics.UUID_IDENTIFIER, ModelStatics.SUBJECT_IDENTIFIER, ModelStatics.PREDICATE_IDENTIFIER, ModelStatics.OBJECT_IDENTIFIER, ModelStatics.ORDER_IDENTIFIER,
+								ModelStatics.EVIDENCE_IDENTIFIER, ModelStatics.CONFIDENCE_IDENTIFIER, jp.getCurrentToken()), jp.getCurrentLocation());
+				}
+			}
+		}
+
+		if (jp.getCurrentToken() == JsonToken.END_OBJECT) {
```

Comment:
the while loop should ensure that the currentToken is END_OBJECT, otherwise it wouldn't stop.